src/main/java/org/sonarsource/sonarlint/ls/EnginesFactory.java

```diff
@@ -20,31 +20,15 @@
 package org.sonarsource.sonarlint.ls;
 
 import java.nio.file.Path;
-import java.util.Collection;
 import java.util.EnumSet;
 import java.util.List;
 import java.util.Set;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicReference;
-import javax.annotation.Nullable;
-import org.sonarsource.sonarlint.core.analysis.api.ClientModulesProvider;
-import org.sonarsource.sonarlint.core.client.legacy.analysis.EngineConfiguration;
-import org.sonarsource.sonarlint.core.client.legacy.analysis.SonarLintAnalysisEngine;
-import org.sonarsource.sonarlint.core.client.utils.ClientLogOutput;
 import org.sonarsource.sonarlint.core.rpc.protocol.common.Language;
-import org.sonarsource.sonarlint.ls.backend.BackendServiceFacade;
-import org.sonarsource.sonarlint.ls.log.LanguageClientLogOutput;
-
-import static java.lang.String.format;
-import static org.sonarsource.sonarlint.ls.util.Utils.interrupted;
 
 public class EnginesFactory {
```

Comment:
🔥 with what is left from this class, I suggest removing the rest of the non-static stuff (e.g. `shutdown`), and renaming it - `EnabledLanguages` maybe?

Also, there is a lot of code related to engines in `ProjectBindingManager`, seems that it can also be cleaned up.